common/plugins/org.jboss.tools.common.el.core/src/org/jboss/tools/common/el/core/ca/AbstractELCompletionEngine.java

```diff
@@ -161,8 +166,11 @@ public IRelevanceCheck createRelevanceCheck(IJavaElement element) {
 	 * @see org.jboss.tools.common.el.core.resolver.ELResolver#resolve(org.jboss.tools.common.el.core.resolver.ELContext, org.jboss.tools.common.el.core.model.ELExpression)
 	 */
 	public ELResolution resolve(ELContext context, ELExpression operand, int offset) {
-		List<Var> vars = new ArrayList<Var>();
+		List<Var> vars = EMPTY_VARS;
 		Var[] array = context.getVars(offset);
+		if(array.length > 0) {

```

Comment:
My suggested line does not use the asList directly, its wrapped in new ArrayList to avoid this issue.

Point was simply that your manual iteration of the array is exactly what asList does anyway.

And yeah, unnecessary objects can be an issue too but I don't know the code enough to tell.

Just spotted the many repeated for loops which seemed unnecessary and easily fixed.